asserts/ifacedecls.go

```diff
@@ -556,10 +646,36 @@ type constraintsDef struct {
 	invert bool
 }
 
-type subruleCompiler func(context string, def constraintsDef) (constraintsHolder, error)
+type subruleContext struct {
```

Comment:
It would be easier to understand if the introduction of this context was a separate patch and if it was documented in any way. I pieced together how I understand this is operating now but I think it would help others if there was a note about what it does.

Reply:
I'll try to add a comment when rebasing the follow up.